AffirmSDK/AffirmCheckoutData.h

```diff
@@ -202,8 +202,8 @@ NS_ASSUME_NONNULL_BEGIN
 /// Additional metadata for the checkout. Optional.
 @property(nonatomic, copy, readonly, nullable) NSDictionary *metadata;
 
-/// The total purchase amount. Dynamically computed from the other properties of the checkout.
-@property(nonatomic, copy, readonly) NSDecimalNumber *total;
+/// The total purchase amount in cents. Optionally dynamically computed from the other properties of the checkout.
+@property(nonatomic, copy, readonly, nullable) NSDecimalNumber *totalAmount;
```

Comment:
We don't want to change the units of an existing property. `totalAmount` is currently returned in dollars and you're changing it to represent cents. This change isn't backwards compatible so anyone currently using this value will need to know to fix this once they update to this version. Ideally we don't make any non backwards compatible changes like this unless we really need to. 

Also, if this is a computed property then it shouldn't be nullable.

I'm a little confused here. Why would you want to pass in the total amount sometimes the other times it would be computed? How would those values differ? Maybe there is a slight difference and we should have a new property instead?

Reply:
As I understand it, this ticket is aimed at accomplishing the following: today, the total price we send to the checkout endpoint is calculated by the SDK as `sum([item.price for item in items]) + taxAmount + shippingAmount - sum([discount.amount for discount in discounts])`. That requires the merchant to separate out their price into item prices, discount amounts, shipping, and tax. This is tough to do in certain cases, and moreover it doesn't match what we do in affirm.js.

This change allows the merchant to pass in a single price, `totalAmount` (in cents), and use that instead of needing to pass in all of the different amounts separately. The total amount we pass to checkout becomes `totalAmount`, not `total`.

Reply:
Hmmm I don't think this change impact compatibility. The one thing we need to be careful is we still need to support the calculation from the Total function to be backward compatible.
[https://github.com/Affirm/affirm-ios-sdk/blob/9388e165255997ffbd058f8cdbf7e0138713f2eb/AffirmSDK/AffirmCheckoutData.m#L280](url)

1. The total variable from the header file is expose but never actually used anywhere.
2. Total is a private function.
3. We say nullable because we want to be backward compatible since our previous merchant is using this automated way to calculate the total already.
4. We always had send in the end the cent prior to this pull request.
[https://github.com/Affirm/affirm-ios-sdk/blob/master/AffirmSDK/AffirmCheckoutData.m#L274](url)